bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gnirs/GnirsRecipe.java

```diff
@@ -62,16 +67,34 @@ public ItcImagingResult serviceResult(final ImagingResult r) {
     }
 
     public ItcSpectroscopyResult serviceResult(final SpectroscopyResult r, final boolean headless) {
-        final List<SpcChartData> dataSets = new ArrayList<SpcChartData>() {{
-            if (instrument.XDisp_IsUsed()) {
-                if (!headless) add(createGnirsSignalChart(r));
-                add(createGnirsS2NChart(r));
-            } else {
-                if (!headless) add(Recipe$.MODULE$.createSignalChart(r, 0));
-                add(Recipe$.MODULE$.createS2NChart(r));
```

Comment:
I just realized that I dropped this check of the `headless` parameter.  I should probably put that back in, but I need to figure out why it only controls one plot and not both of them...